sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -141,27 +152,36 @@ Azure::Core::Credentials::AccessToken CloudShellManagedIdentitySource::GetToken(
     Azure::Core::Credentials::TokenRequestContext const& tokenRequestContext,
     Azure::Core::Context const& context) const
 {
-  return TokenCredentialImpl::GetToken(context, [&]() {
-    using Azure::Core::Url;
-    using Azure::Core::Http::HttpMethod;
-
-    std::string resource;
+  std::string scopesStr;
+  {
+    auto const& scopes = tokenRequestContext.Scopes;
+    if (!scopes.empty())
     {
-      auto const& scopes = tokenRequestContext.Scopes;
-      if (!scopes.empty())
+      scopesStr = TokenCredentialImpl::FormatScopes(scopes, true);
+    }
+  }
+
+  return TokenCache::GetToken(std::string(), GetClientId(), GetAuthorityHost(), scopesStr, [&]() {
```

Comment:
Similar comment - every time I see a wildcard lambda capture I get a titch worried. In this case, I understand that the capture is safe, but I'm worried that it's going to sweep up more than anticipated (when the compiler transforms the lambda into an anonymous class, all the captured values are copied into the anonymous class, which can potentially capture more than is anticipated).

Reply:
It is actually `[this, &context, &scopesStr]`. From `this`, we use `m_tokenCredentialImpl` and `m_requestBody`. I can add it.
But in my opinion the only concern here is whether the lambda is going to outlive any of the values referenced. Since it is byref, it is technically possible, unless you know that the lambda is not being saved anywhere. Whether the reader sees `[&]` or `[this, &context, &scopesStr]`, it does not help to answer that question.
Correct me if you have different perception, I certainly can add it.

Reply:
BTW, I will also add comment about the lifetime in my next commit. I hope it will be helpful to the reader to give that clarity that Jinming and you probably wanted initially.